.github/workflows/tests.yml

```diff
@@ -202,6 +202,18 @@ jobs:
   return-tests:
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        config:
+          - { loc: "tests/returntypes/jax", suite: 'jax'}
+          - { loc: "tests/returntypes/tf", suite: 'tf'}
+          - { loc: "tests/returntypes/autograd", suite: 'autograd'}
+          - { loc: "tests/returntypes/gradients", suite: 'gradients'}
+          - { loc: "tests/returntypes/transforms", suite: 'transforms'}
+          - { loc: "tests/returntypes/paramshift", suite: 'paramshift'}
+          - { loc: "tests/returntypes/finitediff", suite: 'finitediff'}
+          - { loc: "tests/returntypes/others", suite: 'others'}

```

Comment:
I don't think `others` is a descriptive name enough and it's creating an opaque container of test files - it may serve us in the short term, but I fear it remains as such and will be accepted as the main name. I think we should mirror the `pennylane/module_name` sort of convention in this suite too, unless a deviation is very much understandable (e.g., having `autograd` is simpler and likely more intuitive than creating an `interfaces` folder which then either has an `autograd` subfolder or related test files).

I think we'd better simply leave the files that don't belong in a folder in the root `returntypes` folder.